### PR TITLE
Support Pagination for the Websites

### PR DESCRIPTION
**Pull request by Karthikeyantestpress**

### CODE REVIEW

File: mysite/blog/views.py [hunk 1]
```diff
@@ -1,10 +1,23 @@
 from django.shortcuts import render, get_object_or_404
 from .models import Post
+from django.core.paginator import Paginator, EmptyPage, PageNotAnInteger
 
 
 def post_list(request):
-    posts = Post.published.all()
-    return render(request, "blog/post/list.html", {"posts": posts})
+    object_list = Post.published.all()
```
Comment (waranlogesh) [inline mysite/blog/views.py:7]:
```suggestion
    published_posts = Post.published.all()
```

File: mysite/blog/views.py [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
+        posts = paginator.page(page)
+    except PageNotAnInteger:
+        # If page is not an integer deliver the first page
+        posts = paginator.page(1)
+    except EmptyPage:
+        # If page is out of range deliver last page of results
+        posts = paginator.page(paginator.num_pages)
+    return render(
+        request, "blog/post/list.html", {"page": page, "posts": posts}
+    )
```
Comment (waranlogesh) [inline mysite/blog/views.py:20]: Multiple levels of abstraction violation
Extract the pagination logic to a helper function and use like this here `posts = paginate(queryset=posts, per_page=3)`

**Review by waranlogesh — CHANGES_REQUESTED** [review]

**Comment by waranlogesh** [comment]
The title should be `Add pagination support in posts list`

**Comment by Karthikeyantestpress** [comment]
New PR for this : https://github.com/Karthikeyantestpress/Blog_App/pull/13

**Pull request closed without merging**